docs/diagrams/wallet-lifecycle/wallet-lifecycle-1-handling-frauds.tikz

```diff
@@ -0,0 +1,187 @@
+% TIP: To generate `.pdf` file out of this `.tikz` file run:
+% `pdflatex filename.tikz`.
+% To install `pdflatex` on MacOS either run `brew install --cask mactex` (not
+% recommended, downloads 4GB of data) or install `basictex` distribution plus
+% needed packages (recommended, downloads only essential packages):
+% ```
+% brew install --cask basictex
+% sudo tlmgr update --self
+% sudo tlmgr install standalone
+% sudo tlmgr install comment
+% pdflatex <filename>.tikz
+%```
+% 
+% To generate `.png` file out of `.pdf` on MacOS run:
+% `convert -density 300 <filename>.pdf -resize 50% <filename>.png`.
+% This requires installation of the `imagemagick` tool:
+% `brew install imagemagick`.
+
+\documentclass{standalone}%
+\usepackage{tikz}%
+\usetikzlibrary{positioning,shapes,arrows.meta,chains}%
+
+% These commands are used to draw arrow labels
+\newcommand{\yes}{node [auto, pos=0, xshift=5mm]{Y}}%
+\newcommand{\no}{node [auto, swap, pos=0, xshift=-5mm]{N}}%
+
+% We need layers to draw the background
+\pgfdeclarelayer{background}%
+\pgfsetlayers{background,main}%
+
+\begin{document}%
+\begin{tikzpicture}[
+% Specify styles for nodes:
+    base/.style = {rectangle, rounded corners, draw=black,
+                   minimum width=4cm, minimum height=1cm,
+                   text centered, text width=4cm, align=flush center},
+   start/.style = {base, fill=blue!30},
+  revert/.style = {base, fill=red!30},
+  action/.style = {base, fill=black!5},
+   check/.style = {diamond, aspect=1.2, draw=black, fill=black!5,
+                   minimum width=1cm, align=center},
+   note1/.style = {rectangle, draw, dashed, fill=green!10,
+                   align=flush left, text width=5cm},
+   note2/.style = {rectangle, draw, dashed, fill=cyan!10,
+                   align=flush left, text width=5cm},
+              > = Triangle,                  % defines the arrow tip style when `->` is used
+           font = \sffamily,
+]
+
+% FRAUD CHALLENGE
+% Nodes
+\node[start]                                                     (fc_start)   {Send \textbf{fraud challenge}};
```

Comment:
We sometimes name the start action of the diagram as *Send ...*, sometimes *Somebody notifies...*, sometimes *Sombody sends...*. I think it makes sense to align them and use `Bridge` function names. This way, it should be easier to locate the given code path on the diagram.

In case of the frauds diagram, we would have:
- Submit fraud challenge,
- Defeat fraud challenge,
- Defeat fraud challenge with heartbeat,
- Notify fraud challenge defeat timeout.

This comment applies to all diagrams.